src/assets/js/admin/settings/common/setupDynamicTemplateFields.js

```diff
@@ -15,6 +15,13 @@ import { toggleFontAppearance } from '../pdf/toggleFontAppearance'
 export function setupDynamicTemplateFields () {
   /* Add change listener to our template */
   $('#gfpdf_settings\\[template\\]').off('change').on('change', function () {
+    const existingSpinner = document.querySelector('.gfpdf-spinner-template')
+
+    /* Remove existing spinner to avoid duplication */
```

Comment:
@debbieellis7 This seems like it's only masking the problem and not actually stopping the double AJAX request. That double request would still be made, but only one spinner would be shown.

Reply:
Yes, This will do for now @jakejackson1. We need to convert to react so it can access our redux store and we will need a discussion for that one first.

Reply:
Instead of removing the existing spinner, can you just return early and not do anything if there's already a request in progress?

Reply:
Yes, I think it will work. Kindly check the latest pushed code @jakejackson1. Thanks